test/unit/custom_spec.js

```diff
@@ -107,3 +107,97 @@ describe("custom canvas rendering", function () {
       .catch(done.fail);
   });
 });
+
+describe("alternate document context", function () {
+  const FontFace = global.FontFace;
+
+  let CanvasFactory;
+
+  beforeAll(() => {
+    global.FontFace = class {};
+    CanvasFactory = isNodeJS ? new NodeCanvasFactory() : new DOMCanvasFactory();
+  });
+
+  afterAll(() => {
+    global.FontFace = FontFace;
+    CanvasFactory = null;
+  });
+
+  it("should use given document for loading fonts", async function () {
+    const domDoc = { fonts: new Set() };
+    const getDocumentParams = buildGetDocumentParams(
+      "TrueType_without_cmap.pdf",
+      {
+        disableFontFace: false,
+        ownerDocument: domDoc,
+      }
+    );
+
+    const task = getDocument(getDocumentParams);
+    const doc = await task.promise;
+
+    const page = await doc.getPage(1);
+
+    const viewport = page.getViewport({ scale: 1 });
+    const canvasAndCtx = CanvasFactory.create(viewport.width, viewport.height);
+
+    await page.render({
+      canvasContext: canvasAndCtx.context,
+      viewport,
+    }).promise;
+
+    expect(domDoc.fonts.size).toBe(1);
+
+    await doc.destroy();
+    await task.destroy();
+    CanvasFactory.destroy(canvasAndCtx);
+    expect(domDoc.fonts.size).toBe(0);
+  });
+
+  it("should use given document for loading fonts (legacy path)", async function () {
```

Comment:
It could be that I'm missing a bit of knowledge about the font loader here, but what legacy path is meant here?

Reply:
There are two paths. One path adds `FontFace` instances directly to `document.fonts` and the other creates a `<style>` element and appends `@font-face` rules to it.

Reply:
Probably clearer to call this test `should use given document for loading fonts (with CSS rules)`.